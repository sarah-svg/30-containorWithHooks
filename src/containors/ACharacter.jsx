import React, { useState, useEffect } from 'react';
import CharacterDetail from '../components/characters/CharacterDetail.jsx';
import { getOneCharacter } from '../services/fetchApi.js';
import { useParams } from 'react-router-dom';

function ACharacter() {
  const [loading, setLoading] = useState(true);
  const [characters, setCharacters] = useState([]);
  const { id } = useParams(); 
  useEffect(() => {
    
    getOneCharacter(id).then((characters) => {
    
      setCharacters(characters);
      setLoading(false);
      console.log(characters, 'hello from the a character page');
      
    });
  }, []);
  if(loading)
    return <h1  data-testid="loading">Loading A Character</h1>;
  return <CharacterDetail {...characters} />;
  // return(<h1>hello</h1>);
}

export default ACharacter;
