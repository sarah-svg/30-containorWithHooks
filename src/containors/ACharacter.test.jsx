import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import ACharacter from './ACharacter';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import  character from '../fixtures/character.json';

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/2', (req, res, ctx) => {
    return res(ctx.json(character));
  })
);

describe('Containor for just one character display', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());
  it('test the loading behavior', () => {
    // eslint-disable-next-line max-len
    render(<MemoryRouter><ACharacter match={{ params: { id: '2' } }}/></MemoryRouter>);

    screen.getByTestId('loading');
    return waitFor(() => {
      screen.getByText('Morty Smith');
      screen.getByText('Human');
      screen.getByText('Alive');
    });
  });
});
