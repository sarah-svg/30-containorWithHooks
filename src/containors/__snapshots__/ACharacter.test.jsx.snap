// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containor for just one character display creates a snapshot test of the character containor 1`] = `
<DocumentFragment>
  <h1
    data-testid="loading"
  >
    Loading A Character
  </h1>
</DocumentFragment>
`;
